src/OrchardCore.Modules/OrchardCore.Taxonomies/Indexing/TaxonomyIndex.cs

```diff
@@ -37,7 +38,7 @@ public override void Describe(DescribeContext<ContentItem> context)
             context.For<TaxonomyIndex>()
                 .Map(contentItem =>
                 {
-                    if (!contentItem.IsPublished())
+                    if (!(contentItem.IsPublished() || contentItem.Latest))
```

Comment:
Need to think about it. If the index was meant to only contain the published version, that would be wrong.

Reply:
The index only contained the Published version. I changed that, because we need to get the Draft version for the ordering page. Every preexisting use of the index was reviewed, to make sure that it is taking the Published version only.

Reply:
I think this is problematic.

The index was definitely designed to only take published.

This breaks around 50% of my queries, but that I can live with.

For example
Published : TermA
Draft: TermC & TermB

Would produce three index records
FooId, TermA
FooId, TermB
FooId, TermC

So FooId is now categorized with TermC, even though it's only in draft, and a query on TermC will return FooId

I think the index would have to also contain values for published and latest, which is what makes it more breaking

Reply:
Yes, I understand. Looks like I disregarded the code that already exists outside the “official” Orchard Core solution.

It's sad that the index was made for Published version only, from the start. Doesn't seem to be the case with the ContainedPartIndex. As you know, with Lists we can order contained Items that are not published yet, and the index has both the Latest and the Published versions (even without those two extra columns).

Should I go back to the original index (plus the “Order” column), and give up on the ability to order Draft versions of categorized content items (only keeping eventual drafts in sync with its published versions)?

Occurred to me creating a second index with both the published and latest versions, and the extra “Published”, “Latest” and “Order” columns, to be used only when Ordering is enabled. But didn't sound too good. Maybe there is a better (non-breaking) way...

(My insistence on being able to order Draft versions comes from my user base. They will want to create a draft and move it to the intended position, before publishing it - instead of having the item appear at the top for a while, just because it’s new. But I can always build an “in-house” solution for my particular use case. No problem.)

Another thing that could change is the pagination. I saw that demo a few days ago where, instead of an option for the “page size”, there was a dropdown at the bottom of the page with several options for the page size. I think that is a better solution for this “ordering page” too.

Reply:
I went a step back and gave up on the ability to order content items that don’t have a published version. This way, there’s no need to make breaking changes to the index (we stick to published items only, on the index).

Even so, if a content item also has a draft (categorized with the same term), the draft order is kept in sync with the moved published version.

There was another problem that needed a solution. When an item is cloned, or unpublished and later published again (etc.), it’s Order value regarding a Term can already be in use by another content item. That would result in duplicate Order values. Now this is detected and solved (see `EnsureUniqueOrderValues`).

(Among other minor improvements, I simplified the `OrderCategorizedContentItems` on the `AdminController`. It doesn’t do `BuildDisplayAsync` anymore, which was confusing and uncalled for.)